api-matlab/wrapper_physiobatch.m

```diff
@@ -0,0 +1,111 @@
+     elseif strcmp(data_vendor, 'GE')
+        [cardiac, ~] = spm_select(1,'any','Specify cardiac file (starting with ECGData_)',{}, default_path);
+        % cardiac = input('Specify file path to the cardiac file (in template starting with ECGData_)', 's');
+        [respiration, ~] = spm_select(1,'any','Specify respiration file (starting with RespData_)',{}, default_path);
+        % respiration = input('Specify file path to the respiration file (in template starting with RespData_)', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {cardiac};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {respiration};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''};
+     elseif strcmp(data_vendor, 'Siemens_HCP')
+        [physiolog, ~] = spm_select(1,'any','Specify physio data file (sth like tfMRI_MOTOR_LR_Physio_log.txt)',{}, default_path);
+        % physiolog = input('Specify file path to the tfMRI_MOTOR_LR_Physio_log.txt file', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {physiolog};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {physiolog};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''};
+     elseif strcmp(data_vendor, 'Siemens')
+        [physiolog, ~] = spm_select(1,'.*.ecg','Specify physio data file (siemens_PAV.ecg)',{}, default_path);
+        % physiolog = input('Specify file path to the siemens_PAV.ecg file', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {physiolog};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {''};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''}; 
+     elseif strcmp(data_vendor, 'Siemens_Tics')
+        [cardiac, ~] = spm_select(1,'.*.log','Specify cardiac file (..._PULS.log)',{}, default_path);
+        % cardiac = input('Specify file path to the cardiac file (in template ending with _PULS.log)', 's');
+        [respiration, ~] = spm_select(1,'.*.log','Specify respiration file (..._RESP.log)',{}, default_path);
+        % respiration = input('Specify file path to the respiration file (in template ending with _RESP.log)', 's');
+        [timing, ~] = spm_select(1,'.*.log','Specify scan timing file (..._Info.log)',{}, default_path);
+        % timing = input('Specify file path to the scan timing file (in template ending with _Info.log)', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {cardiac};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {respiration};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {timing};
+    else
+        ME = MException('error:wronginput', 'Wrong vendor specified. Please specify one of BIDS, Philips, Biopac_Txt, GE, Siemens, Siemens_HCP, Siemens_Tics');
+        throw(ME);
+    end
+    
+    % read scan parameters
+    input_params = inputdlg({'Number of scans:','Number of slices:','Number of dummies:', 'TR:', 'Onset slice:'}, 'Parameters for physio');
```

Comment:
It seems like some of these parameters can be derived from the data that were already analysed at this point, or entered by the user at an earlier stage, e.g. slices, scans, dummies. For clarity, @NinaWie did you create this input dialog because you are uncertain which variables already exist in the workspace, or for a different reason?

Another question, how are number of scans and dummies differentiated? e.g. if I have 5 dummies and 210 scans, that totals 215 scans. Is the "Numer of scans" then 210 or 215?

Reply:
I think the number of dummies is in addition to the others, so the number of scans would be 210. Good point, we have to specify this in the input dialog I guess. Regarding the other parameters, I couldn't find any except the number of scans in the previous parameters when doing online processing with rtQC, and I asked Lydia about it as well. We could read the number of slices and number of scans from the header of the nifti file, but since other parameters need to be input manually, we thought it would it better to do that for all of these parameters, also in case that a user processes some data online and then wants to do physio processing on other data. We can also discuss that in a Skype though.

Reply:
Ok, so as discussed you'll add most of the fields required for PhysIO to the defaults m-file. Then on the Post QC tab only ask the user to input the necessary remaining files/parameters (e.g. the physio traces as you do currently). Do you have the same understanding?

What I could do after that is run through the parameters you've added to the m-file and double check that they are not already declared or generated somewhere else throughout the process of Pre-QC and Online QC.